### PR TITLE
Issue/federated

### PR DESCRIPTION
**Pull request by sosandstrom**
Merga inte denna, det är strul bara. Får inte till federated-flowet ännu.

### CODE REVIEW

**Comment by sosandstrom** [comment]
ready for review and hopefully merge

File: guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java [hunk 1]
```diff
@@ -83,28 +84,30 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
...
       final DConnection conn = verifyAccessToken(accessToken);
-      if (null != conn) {
-        LOGGER.info("Authenticated");
-        LOGGER.debug(" userId = {} roles {}", conn.getUserId(), conn.getRoles());
 
-        // User is authenticated
-        request.setAttribute(NAME_CONNECTION, conn);
-        request.setAttribute(NAME_USER_ID, conn.getUserId());
-        request.setAttribute(NAME_ROLES, conn.getRoles());
+      // access_token used here for app authentication must be issued by self:
```
Comment (mattiaslevin) [inline guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java]: Not sure I understand this.
How are roles etc handled if other provider then self?

**Comment by mattiaslevin** [comment]
I understand that you do not want the federated request to pass through the BA filter.

Not sure I understand the changes in OAuth filter.
How is roles set handled for other providers?

File: guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java (hunk 1 above)
Comment (sosandstrom) [inline guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java]: They're not! A federated token must only be used when registering to get hold of the application-specific token. It must not be used subsequently to access the resources.

Comment (mattiaslevin) [inline guja-core/src/main/java/com/wadpam/guja/oauth2/web/OAuth2Filter.java]: Ahh, I understand.
How does the federated token gets traded for an internal token? How is the association made?
(lets have a chat later)

**Pull request merged**